.markdownlint.yaml

```diff
@@ -12,13 +12,35 @@
 # limitations under the License.
 
 # markdownlint-cli configuration file.
+# To run markdownlint, use the following bash script:
+#   bash build_tools/github_actions/lint_markdown.sh <file>
 
-# Enable all rules by default
+# Enable all markdownlint rules by default
 default: true
 
-# Prefer indented lists for legacy compatibility
+# List formatting:
+# 1. Lists must have a blank line before AND after the list.
+# 2. Lists start aligned to the left (do not indent the top level list items).
+# 3. You may use one or zero blank lines between list items.
+# 4. Nested list items must be indented to align with the first character of
+#    the first line. For bullet lists, that means 2 spaces. For numbered
+#    lists, that's 3 spaces (but 4 spaces is okay if that's easier).
+# 5. In multiline list items, subsequent lines are indented by 2 spaces.
+#    This is not checked automatically, so we're documenting this convention
+#    to make sure the codebase stays consistent.
+#
+# Examples:
+#
+# * This is a list item that has multiple
+#   lines and each line aligns with the text from the first line.
+#   * This is a nested list, also aligned with the first line.
+#
+# For ordered lists, that means three spaces for wrapped lines:
+#
+# 1. This is an ordered list item.
+#    1. The nested list aligns with the first line.
 ul-indent:
-  start_indented: true
```

Comment:
Perhaps let's explicitly call out `start_indented: false`?

Reply:
I'd prefer not. There are a lot of default properties like this ([see them all here](https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.yaml), so I think it's best to just keep our config limited to the overrides.

Reply:
Oops. I'm a hypocrite. `indent: 2` is also the default, but if we don't have that rule declared here then we have no context for our style guide for list items. So I think we should keep that one here until we have another document where we describe all the Markdown style guidelines such as these list styles.